design_docs/config-file_reorganization.md

```diff
@@ -0,0 +1,132 @@
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+There should be a simple, intuitive method for turning on and off individual analysis modules (e.g. `ocean/ohc_timeseries`).  This should replace the current approach of having a boolean `generate` flag for each analysis module in a separate config section.  Preferably, there should be an equivalent method for turning on and off analysis modules from the command line that overrides that in the config file.
+
+<h2> Requirement: there should be a simplified template for config files <br>
+Date last modified: 2017/02/01 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The current example config file should be made into a general template.  Simplifications should be made to the template so that it can more easily and intuitively be modified for several analyses.  Example config files should also be added for analyzing several existing runs on several different machines.
+
+<h2> Requirement: removal of ACME specific config options <br>
+Date last modified: 2017/02/01 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+To the extent possible, ACME-specific config options such as `casename` and `ref_casename_v0` should be generalized in a way that is also appropriate not just ACME runs but also any other runs involving the MPAS components we support.
+
+<h2> Requirement: consistent section and option names <br>
+Date last modified: 2017/02/01 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+A consistent convention of capitalization and underscores should be used throughout the config file.
+
+
+<h1> Design and Implementation </h1>
+
+<h2> Implementation: a simple way of turning on and off individual analysis modules <br>
+Date last modified: 2017/02/02 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The following comment describes the planned implementation in the config file.
+```
+# a list of analyses to generate.  Valid names are:
+#   'plotTimeSeriesOHC', 'plotTimeSeriesSST', 'plotRegriddedSST', 
+#   'plotRegriddedSSS', 'plotRegriddedMLD', 'plotTimeSeriesSeaIce', 
+#   'plotRegriddedSeaIce'
+# the following shortcuts exist:
+#   'all' -- all analyses will be run
+#   'all_plotTimeSeries' -- all time-series analyses will be run
+#   'all_plotRegriddedHorizontal' -- all analyses involving regridded horizontal
+    #                                fields will be run
+#   'all_ocean' -- all ocean analyses will be run
+#   'all_seaIce' -- all sea-ice analyses will be run
+#   'no_plotTimeSeriesOHC' -- skip 'plotTimeSeriesOHC' (and similarly with the
+#                             other analyses).
+#   'no_ocean', 'no_plotTimeSeries', etc. -- in analogy to 'all_*', skip the 
+#                                            given category of analysis
+# an equivalent syntax can be used on the command line to override this
+# option:
+#    ./run_analysis.py config.analysis --generate \
+#         all,no_ocean,all_plotTimeSeries
+generate = ['all']
```

Comment:
This demonstrates how I'd like to rename the different analysis scripts using CamelCase.  @pwolfram and @milenaveneziani, again, I want to make sure you're good with this before I dive into implementation.